pkgs/by-name/me/megapixels/package.nix

```
@@ -27,15 +27,15 @@ let
     ++ optionals jpgSupport [ graphicsmagick exiftool ]
   );
 in
-stdenv.mkDerivation rec {
-  pname = "megapixels";
-  version = "1.7.0";
+stdenv.mkDerivation (finalAttrs: {
+  pname = "Megapixels";
```

Isn't
```suggestion
  pname = "megapixels";
```
the convention?

---

From https://github.com/NixOS/nixpkgs/blob/c9dda2db28825cb690dc647b74c6cd7dde795d89/pkgs/README.md:
"The `pname` attribute _should_ be identical to the upstream package name."
But you're right, I can't follow that advice since the next sentence prohibits uppercase letters:
"The `pname` and the `version` attribute _must not_ contain uppercase letters — e.g., `"mplayer"` instead of `"MPlayer"`."
I traced it back to eb9ff07031d8b1845d6d84579d1b19481977e319 where Eelco added in 2009:
"Don’t use uppercase letters in the `name` attribute — e.g., `"mplayer-1.0rc2"` instead of `"MPlayer-1.0rc2"`."
What I'm missing though is an explanation of _why_ this is important. Any ideas?
In my personal opinion it is more important to follow the upstream name than to change the package name to conform to the convention.
There are also many packages that don't follow this rule, for example the R package or SDL.
Anyways, I changed `pname` back to lowercase.
Instead I decided to rename the package attribute name and add an alias since I couldn't find any rules regarding the package attribute name. What are your opinions on that change?

---

> What I'm missing though is an explanation of _why_ this is important. Any ideas?

It makes it easier to guess it and it is more uniform. Also some projects are not very consistent. 

> There are also many packages that don't follow this rule, for example the R package or SDL.

Those should really be renamed but for packages like SDL that is usually a bigger undertaking.

> What are your opinions on that change?

Not sure if there are explicit rules about that but generally pname should be equal to attr unless eg we have a wrapper or some other good reason.

---

Okay. I'll wait for @dotlambda's feedback too and then probably rename it back.

---

> generally pname should be equal to attr

I agree.